pkg/katautils/config.go

```diff
@@ -918,6 +931,18 @@ func checkFactoryConfig(config oci.RuntimeConfig) error {
 		}
 	}
 
+	if config.FactoryConfig.VMCacheNumber > 0 {
+		if config.HypervisorType != vc.QemuHypervisor {
+			return errors.New("VM cache just support qemu")
+		}
+		if config.AgentType != vc.KataContainersAgent {
+			return errors.New("VM cache just support kata agent")
+		}
+		if config.HypervisorConfig.UseVSock {
+			return errors.New("config vsock conflicts with VM cache, please disable one of them")
```

Comment:
It does not really conflict. Need to send the guest vsock cid/port over cache server grpc. Maybe a todo in future PR.